nilearn/image/resampling.py

```diff
@@ -340,6 +339,8 @@ def resample_img(img, target_affine=None, target_shape=None,
     This function handles gracefully NaNs and infinite values in the input
     data, however they make the execution of the function much slower.
     """
+    from .image import new_img_like
```

Comment:
> > +    from .image import new_img_like

> making these imports global gives 15 errors due to circular imports, too

This tells me that new_img_like should probably be defined in utils, and
imported in places where it's used (including image to expose it).

Other option is to always import it from image.image in the codebase. I
think that this might solve the circular import problems.